src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -195,7 +195,9 @@ public object GetProperyValue(object obj, string propertyName)
                 var completed = await CompleteValue(context, context.Schema.FindType(nonNullType.Type), fields, result);
                 if (completed == null)
                 {
-                    throw new ExecutionError("Cannot return null for non-null type. Field: {0}".ToFormat(nonNullType.Name));
```

Comment:
`nonNullType.Name` was consistently `null`, and was referring to the type name rather than the field name anyway. So think this makes more sense, at least from the testing that I've gone through.